pkgs/tools/filesystems/ceph/default.nix

```diff
@@ -166,7 +169,74 @@ let
 
   # Watch out for python <> boost compatibility
   python = python310.override {
-    packageOverrides = self: super: {
+    packageOverrides = self: super: let cryptographyOverrideVersion = "40.0.1"; in {
+      # Ceph does not support `cryptography` > 40 yet:
+      #     https://github.com/NixOS/nixpkgs/pull/281858#issuecomment-1899358602
+      # We pin the older `cryptography_40` here;
+      # this also forces us to pin an older `pyopenssl` because the current one
+      # is not compatible with older `cryptography`, see:
+      #     https://github.com/pyca/pyopenssl/blob/d9752e44127ba36041b045417af8a0bf16ec4f1e/CHANGELOG.rst#2320-2023-05-30
+      cryptography-vectors = (self.callPackage ../../../development/python-modules/cryptography/vectors.nix { }).overridePythonAttrs (old: {
+        src = self.fetchPypi {
+          pname = "cryptography_vectors";
+          version = cryptographyOverrideVersion;
+          hash = "sha256-hGBwa1tdDOSoVXHKM4nPiPcAu2oMYTPcn+D1ovW9oEE=";
+        };
+      });
+      cryptography = super.cryptography.overridePythonAttrs (old:
+        let
+          cryptographySrc = fetchPypi {
+            inherit (old) pname;
+            version = cryptographyOverrideVersion;
+            hash = "sha256-KAPy+LHpX2FEGZJsfm9V2CivxhTKXtYVQ4d65mjMNHI=";
+          };
+        in
+        {
```

Comment:
```suggestion
      cryptography = super.cryptography.overridePythonAttrs (old: rec {
```

Reply:
I'll click "commit suggestion" so I can check if this transformation works.

Reply:
I see, you added `rec`. That works, `.drv` is unchanged. I was scared first to add the `rec` here because if forgotten, it will take `ceph`'s `src`, but I guess one will notice that rather quickly.